app/common/renderer/components/Session/AttachToSession.jsx

```diff
@@ -1,46 +1,83 @@
 import {ReloadOutlined} from '@ant-design/icons';
-import {Button, Card, Col, Form, Row, Select} from 'antd';
+import {Button, Card, Col, Form, Row, Select, Tooltip} from 'antd';
+import _ from 'lodash';
 import React from 'react';
 
 import {ServerTypes} from '../../actions/Session';
 import SessionStyles from './Session.module.css';
 
-const formatCaps = (caps) => {
-  let importantCaps = [caps.app, caps.platformName, caps.deviceName];
-  if (caps.automationName) {
-    importantCaps.push(caps.automationName);
+class DefaultSessionDescription {
+  constructor(caps) {
+    this._caps = caps;
   }
-  return importantCaps.join(', ').trim();
-};
 
-const formatCapsBrowserstack = (caps) => {
-  let importantCaps = formatCaps(caps).split(', ');
-  if (caps.sessionName) {
-    importantCaps.push(caps.sessionName);
+  // sessionName is only populated for cloud providers
+  _fetchSessionName() {
+    return this._caps.sessionName;
   }
-  return importantCaps.join(', ').trim();
-};
 
-const formatCapsLambdaTest = (caps) => {
-  if (caps.hasOwnProperty.call(caps, 'capabilities')) {
-    caps = caps.capabilities;
+  _fetchDeviceInfo() {
+    return this._caps.deviceName || this._caps.avd || this._caps.udid;
   }
-  const deviceName = caps.desired ? caps.desired.deviceName : caps.deviceName;
-  const importantCaps = [deviceName, caps.platformName, caps.platformVersion];
-  return importantCaps.join(', ').trim();
-};
 
-const getSessionInfo = (session, serverType) => {
+  _fetchPlatformInfo() {
+    if (this._caps.platformName) {
+      const platformInfo = this._caps.platformVersion
+        ? `${this._caps.platformName} ${this._caps.platformVersion}`
+        : this._caps.platformName;
+      return platformInfo;
+    }
+  }
+
+  _fetchAutomationName() {
+    return this._caps.automationName;
+  }
+
+  _fetchAppInfo() {
+    return this._caps.app || this._caps.bundleId || this._caps.appPackage;
+  }
+
+  transform() {
```

Comment:
for the given  class names I would rather rename this method to something like `calculate` or `extract`

Reply:
Changed to `assemble`